sql/mysql_db/privileged_database_provider.go

```diff
@@ -43,13 +43,15 @@ func NewPrivilegedDatabaseProvider(grantTables *MySQLDb, p sql.DatabaseProvider)
 
 // Database implements the interface sql.DatabaseProvider.
 func (pdp PrivilegedDatabaseProvider) Database(ctx *sql.Context, name string) (sql.Database, error) {
-	privSet := pdp.grantTables.UserActivePrivilegeSet(ctx)
-	// If the user has no global static privileges or database-relevant privileges then the database is not accessible.
-	if privSet.Count() == 0 && !privSet.Database(name).HasPrivileges() {
-		return nil, sql.ErrDatabaseAccessDeniedForUser.New(pdp.usernameFromCtx(ctx), name)
-	}
-	if strings.ToLower(name) == "mysql" {
-		return pdp.grantTables, nil
+	if strings.ToLower(name) != sql.InformationSchemaDatabaseName {
```

Comment:
We're using `strings.ToLower(name)` twice here, better to do it once, cache it to a variable, and reference that variable in both locations.